en/resources/tutorials/templating.md

```diff
@@ -84,43 +164,154 @@ There are {{ articles.count }} articles.
+`MustacheExample.mustache`
+
+6. Add the following into the `MustacheExample.mustache` file: 
 
+{% raw %}
 ```
-/myTemplateApp
-|
-|--/Sources
-  |--/myTemplateApp
-    |-- main.swift
-|--/Views
-  |-- document.mustache
-  |-- document.stencil
+Hello {{name}}
+Your beard trimmer will arrive on {{format(date)}}.
+{{#late}}
+Well, on {{format(realDate)}} because of a Martian attack.
+{{/late}}
 ```
+{% endraw %}
 
 ---
 
 ## Registering a template engine with a Router instance
-<span class="arrow">&#8227;</span> You should import the package of the template engine:
 
+We now have everything in place to start writing the code for our implementation of the template engines!  
+In Xcode open the `Application.swift` file found in `Sources/Application/`
+
+The first step we do to do is import our packages so we can make use of the API they provide. Below the line: 
+```swift 
+import Health
+```
+
+Add the following: 
 ```swift
 import KituraMustache
 import KituraStencil
 ```
+This will now allow us to use both of the template engines within our code.
 
-<span class="arrow">&#8227;</span> To use the template engine, you must register it with a Router instance. This is done by calling the  `Router.add(templateEngine:)` function, e.g.:
+> ![tip] You may need to build your project before you can import the template engines, this can be done in Xcode using `cmd + B`
+
+To use the template engine, you must register it with a Router instance. This is done by calling the  `Router.add(templateEngine:)` function, e.g.:
+
+In the `postInit()` function below the line: 
 
 ```swift
-let router = Router()
+initializeHealthRoutes(app: self)
+```
+Add the following
+```swift
 router.add(templateEngine: MustacheTemplateEngine())
 router.add(templateEngine: StencilTemplateEngine())
 ```
+We have now registered both of our template engines with a router.
+
 
 ---
 
 ## Rendering a template
-<span class="arrow">&#8227;</span> You can render a template by calling the `response.render(_:context)` function. Here's the code for the `document.mustache` and `document.stencil` template files we referenced earlier:
+We can render a template by calling the `response.render(_:context)` function, where context is the information the template engine will use to fill out the corresponding template files.
+
+We will start with rendering the Stencil template. Below the line:  
+
+```swift
+router.add(templateEngine: StencilTemplateEngine())
+``` 
+
+add the following: 
+```swift
+// Handle HTTP GET requests to Stencil
+router.get("/articles") { request, response, next in
+
+    // The example from https://github.com/kylef/Stencil/blob/master/README.md
+    let context = [
+       "articles": [
+         ["title": "Migrating from OCUnit to XCTest", "author": "Kyle Fuller"],
+         ["title": "Memory Management with ARC", "author": "Kyle Fuller" ]
+       ]
+    ]
+
+    try response.render("StencilExample.stencil", context: context).end()
```

Comment:
I think we should only use one of .end() or next() not both. This applies though the rest of the tutorial.